hub/values.yaml

```diff
@@ -482,15 +482,22 @@ jupyterhub:
               await asyncio.sleep(self.spawn_launcher_delay)
 
             # give admins read/write access to the shared folder
-            volume_mounts = self.volume_mounts
+            volume_mounts = self.volume_mounts.copy()
+
             if self.user.admin:
-              for v in volume_mounts:
-                if v.get('mountPath') == '/home/jovyan/shared':
-                  rw = v.copy()
+              for v in self.volume_mounts:
```

Comment:
You can create a copy of the list, so you don't have to worry about modifying a dict during iteration:

```python
for name, v in list(self.volume_mounts.items()):
    if v.get('mountPath') == '/home/jovyan/shared':
        rw = v.copy()
        rw['readOnly'] = False
        rw['mountPath'] = '/home/jovyan/shared_readwrite'
        rw_name = f"{name}-readwrite"
        self.volume_mounts[rw_name] = rw
        self.log.info(f'Giving {self.user.name} read/write access to shared folder')
```

(the only changes from before is `list(self.volume_mounts.items())` to iterate through a copy instead of mutating the dict while updating it, and `self.volume_mounts[rw_name] = rw` to update a dict instead of a list)